prov/gni/include/gnix_cq.h

```diff
@@ -45,6 +45,9 @@
 #define GNIX_CQ_DEFAULT_FORMAT struct fi_cq_entry
 #define GNIX_CQ_DEFAULT_SIZE   256
 
+/* forward declaration */
+struct gnix_fid_ep;
```

Comment:
I think the preprocessor gets confused because of circular dependencies although I thought that's what include guards were for.